services/QuillLMS/app/workers/teacher_notifications/send_notification_worker.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module TeacherNotifications
+  class SendNotificationWorker
+    include Sidekiq::Worker
+
+    def perform(activity_session_id)
+      @activity_session = find_activity_session(activity_session_id)
+      return unless @activity_session
+
+      TeacherNotification.transaction do
+        send_complete_diagnostic
+        send_complete_all_diagnostic_recommendations
+        send_complete_all_assigned_activities
```

Comment:
Another question. We'll be adding more of these notifications in the future. Is the plan to add them all in this file? That might make testing and maintenance more complicated than breaking them up into separate jobs.